sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_path_client.py

```diff
@@ -67,12 +68,18 @@ def __init__(
         if datalake_hosts:
             blob_primary_account_url = convert_dfs_url_to_blob_url(datalake_hosts[LocationMode.PRIMARY])
             blob_hosts = {LocationMode.PRIMARY: blob_primary_account_url, LocationMode.SECONDARY: ""}
-        self._blob_client = BlobClient(blob_account_url, file_system_name, path_name,
-                                       credential=credential, _hosts=blob_hosts, **kwargs)
-
-        _, sas_token = parse_query(parsed_url.query)
+        path_snapshot, sas_token = parse_query(parsed_url.query)
         self.file_system_name = file_system_name
         self.path_name = path_name
+        try:
+            self.snapshot = snapshot.snapshot # type: ignore
+        except AttributeError:
+            try:
+                self.snapshot = snapshot['snapshot'] # type: ignore
+            except TypeError:
+                self.snapshot = snapshot or path_snapshot
```

Comment:
I think snapshot shouldn't reside here since directory doesn't support snapshot.
we can delete this part and directly use the snapshot in self._blob_client from data_lake_file_client.py